### ISSUE TITLE
Restore checkpoint from iteration

### ISSUE THREAD

**Issue by lucastabelini** [open-mmlab/mmdetection#336]
I modified the code to save a checkpoint every 10k iterations. Is it possible to restore it from the iteration it stopped at? When I tried to restore the checkpoint it started from the iteration 0, even though it was not at iteration 0.

**Comment by hellock** [comment]
It is possible to resume from a checkpoint saved during an epoch. You may refer to the [resume](https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/runner.py#L296) method.

**Comment by lucastabelini** [comment]
I see. But at the moment, this method does not restore the epoch's iteration (inner_iter, in the code). Is there enough information in the checkpoint file to restore that so it resumes from the point where it stopped in the data_loader?

**Comment by hellock** [comment]
You are right, `len(dataloader)` is missing to recover the `inner_iter`, if it is directly provided. Maybe a solution is to save `inner_iter` to the checkpoint.

**Comment by lucastabelini** [comment]
I don't know exactly how the batch loading is implemented, does it take into account which images were already used to train the model so that at each epoch the whole dataset is used? If that's the case, maybe just saving inner_iter to the checkpoint is not the best solution, since with the new random seed we may train with images already used in the epoch. If it does not that take into account, then saving inner_iter and starting the data_loader enumeration from this is enough.

**Comment by hellock** [comment]
Currently resuming from iterations will only retrieve the parameters and optimizer status, but not data orders. That means if inner_iter is set to 1000, the training data will not skip the first 1000 batches automatically.

**Comment by lucastabelini** [comment]
Maybe if I just save inner_iter to the checkpoint and modify [this](https://github.com/open-mmlab/mmcv/blob/c8c3493868e5363e2b118944b199f11e7cb4f0b4/mmcv/runner/runner.py#L257):
```
for i, data_batch in enumerate(data_loader):
```
To:
```
for _ in range(self._inner_iter):  
    next(data_loader) # skip batches already trained  
for i, data_batch in enumerate(data_loader):
```

It might work? It's kind of a hack though.
